From: =?utf-8?q?Christian_G=C3=B6ttsche?= <cgzones@googlemail.com>
Date: Tue, 20 Apr 2021 17:41:10 +0200
Applied-Upstream: https://github.com/logrotate/logrotate/commit/2b588b5ec2e5c27bee857c4abeddafa6a9602ebc
Subject: Log if keyword is not properly separated

The man page states
  Values are separated from directives by whitespace and/or an
  optional =.

But logrotate does accept no separator, like
  rotate7

Log those occurrences with a normal severity, as this usage is not
intended.
---
 config.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/config.c b/config.c
index df2d90b..b0ccdb5 100644
--- a/config.c
+++ b/config.c
@@ -1094,6 +1094,11 @@ static int readConfigFile(const char *configFile, struct logInfo *defConfig)
                     key = isolateWord(&start, &buf, length);
                     if (key == NULL)
                         continue;
+                    if (!isspace((unsigned char)*start)) {
+                        message(MESS_NORMAL, "%s:%d keyword '%s' not properly"
+                                " separated, found %#x\n",
+                                configFile, lineNum, key, *start);
+                    }
                     if (!strcmp(key, "compress")) {
                         newlog->flags |= LOG_FLAG_COMPRESS;
                     } else if (!strcmp(key, "nocompress")) {
